### PR TITLE
Fix publisher

### PR DESCRIPTION
**Pull request by joelrebel**
#### What does this PR do

[fixes inventory publish being skipped on a bios config collection error](https://github.com/metal-toolbox/alloy/commit/3a5a783d3465c78202bd59d5103c16d13b4bef8c)

### CODE REVIEW

File: internal/model/model.go [hunk 1]
```diff
@@ -57,14 +61,24 @@ type Asset struct {
...
 	if a.Errors == nil {
 		a.Errors = map[string]string{}
 	}
 
-	a.Errors[key] = value
+	a.Errors[string(key)] = value
+}
+
+func (a *Asset) HasError(cErr CollectorError) bool {
+	for k := range a.Errors {
```
Comment (DoctorVin) [inline internal/model/model.go:75]: Why not do a map lookup here? Am I missing something?

File: internal/publish/serverservice.go [hunk 2]
```diff
@@ -247,17 +248,40 @@ func (h *serverServicePublisher) publish(ctx context.Context, device *model.Asse
...
+			metricInventorized.With(prometheus.Labels{"status": "failed"}).Add(1)
+			return
+		}
 	}
 
 	// count devices with no errors
+	// TODO: metric label has to be updated to status-inventory, status-bioscfgcollection
 	metricInventorized.With(prometheus.Labels{"status": "success"}).Add(1)
 
+	if !device.HasError(collect.OutofbandInventoryError) {
```
Comment (DoctorVin) [inline internal/publish/serverservice.go:263]: Shouldn’t publishing be all-or-nothing?

File: internal/model/model.go (hunk 1 above)
Comment (joelrebel) [reply internal/model/model.go:75]: :slaps forehead, yeah this was me being stupid

Comment (joelrebel) [reply internal/model/model.go]: fixed in https://github.com/metal-toolbox/alloy/pull/146/commits/a5bcee07d59eb6616cf14353fffccedd6caed746

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**